Makefile

```diff
@@ -21,15 +21,27 @@ mypy:
 
 .PHONY: tests
 tests: 
-	uv run pytest 
+	uv run pytest -m "not integration"
```

Comment:
let me know if we don't want to user markers for this and I'll find another way to test the integration tests. The main requirement for integration tests is docker that is quite prevalent even in github actions but I can understand if you might not want it.